### ISSUE TITLE
False E0608 when indexing an array in a let initializer

### ISSUE THREAD

**Issue by feibyzz** [rust-lang/rust-analyzer#23220]
## Description

rust-analyzer incorrectly reports E0608 for valid array indexing expressions used as `let` initializers.

The same indexing expression does not produce an error when used inside `println!`.

`cargo check` succeeds without any errors.

## Reproduction

```rust
fn main() {
    let a = [1, 2, 3, 4, 5];

    let first = a[0];
    let second = a[1];

    println!("{}", a[1]);
}
````

rust-analyzer reports:

```text
error[E0608]: cannot index into a value of type `[i32; 5]`
```

on both:

```rust
let first = a[0];
let second = a[1];
```

However, this does not produce an E0608 diagnostic:

```rust
println!("{}", a[1]);
```

The type of `a` is correctly shown as `[i32; 5]` when hovering over it in VS Code.

`cargo check` succeeds:

```text
warning: unused variable: `first`
warning: unused variable: `second`
Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.01s
```

The warnings are only because the two variables are unused.

## Environment

* OS: Arch Linux
* rustc: `rustc 1.98.0 (88d9e12ae 2026-08-18) (Arch Linux rust 1:1.98.0-1)`
* cargo: `cargo 1.98.0 (797e8a9bc 2026-08-05) (Arch Linux rust 1:1.98.0-1)`
* rust-analyzer: `rust-analyzer 0.4.3023-standalone`
* VS Code extension: `rust-lang.rust-analyzer@0.4.3023`
* Editor: VS Code

No custom rust-analyzer configuration is required to reproduce the issue.

## Expected behavior

All array indexing expressions should be accepted without an E0608 diagnostic.

## Actual behavior

`a[0]` and `a[1]` produce a false E0608 diagnostic when used as `let` initializers, while the same `a[1]` expression inside `println!` does not.

`cargo check` accepts all of them.

This appears to be related to rust-analyzer issue #22628, although this reproduction does not use `no_std`, a custom target, or `Vec`.

```
```

**Comment by ChayimFriedman2** [comment]
I cannot reproduce this issue using stock toolchain. Do I understand correctly that your toolchain is modified (`(Arch Linux rust 1:1.98.0-1)`)? Do you have rust-src installed?

**Comment by feibyzz** [comment]
> I cannot reproduce this issue using stock toolchain. Do I understand correctly that your toolchain is modified (`(Arch Linux rust 1:1.98.0-1)`)? Do you have rust-src installed?

You were right. I didn't have `rust-src` installed.

I installed the matching `rust-src` package (`1:1.98.0-1`) and restarted VS Code. The false E0608 diagnostics disappeared.

So this issue was caused by the missing `rust-src` package. Sorry for the false report, and thank you for pointing me in the right direction!
